rust/arrow/src/compute/kernels/take.rs

```diff
@@ -423,7 +423,7 @@ where
     let mut offsets_buffer = MutableBuffer::from_len_zeroed(bytes_offset);
 
     let offsets = offsets_buffer.typed_data_mut();
-    let mut values = Vec::with_capacity(bytes_offset);
+    let mut values = MutableBuffer::new(0);
```

Comment:
Used to be here `bytes_offset` but that doesn't make a lot of sense.
The values array could be empty as well (e.g. for empty strings), or have a different capacity altogether.